### PR TITLE
Shooter subsystem

### PR DESCRIPTION
**Pull request by Xmde**

### CODE REVIEW

**Review by owen-mccormick — COMMENTED** [review]
Everything looks good, although I probably wouldn't know if something was wrong. You may want to add a new line at the end of "Shooter.java".

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            D = 0,
+            FF = 0,
+            IZ = 0,
+            MAX_RPM = 0;
+
+    private final CANSparkMax shooterMotor = new CANSparkMax(
+            Ports.SHOOTER_MOTOR,
+            CANSparkMaxLowLevel.MotorType.kBrushless);
+
+    private final DoubleSolenoid hood = new DoubleSolenoid(
```
Comment (django-wardlow) [inline src/main/java/frc/robot/subsystems/Shooter.java]: we are useing a single solenoid, it still gives us 2 positions (true and false), and is much cheaper. the downside and whi we dident use them in the passed is that they dont rember what state they are in when they loose power and always deafult to false

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    public void zeroEncoders() {
+        setEncoderValue(0);
+    }
+
+    /**
+     * Gets the current shooter position in rotations.
+     *
+     * @return The current shooter position
+     */
+    public double getShooterPosition() {
```
Comment (django-wardlow) [inline src/main/java/frc/robot/subsystems/Shooter.java]: you dont need both getEncoderValue() and getShooterPosition()

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+     * Gets the current hood position.
+     * True - Piston Extended
+     * False - Piston Not Extended
+     *
+     * @return The current hood position.
+     */
+    public boolean getHoodPos() {
+        return hoodPos;
+    }
+
```
Comment (django-wardlow) [inline src/main/java/frc/robot/subsystems/Shooter.java:179]: a setHoodPos(bool pos) would be nice for binding to a button

**Review by django-wardlow — COMMENTED** [review]
overall looks good to me, just a few changes and it will be ready to merge

**Pull request merged**